src/test/java/htsjdk/samtools/cram/build/SliceFactoryTest.java

```diff
@@ -0,0 +1,238 @@
+package htsjdk.samtools.cram.build;
+
+import htsjdk.HtsjdkTest;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.cram.CRAMException;
+import htsjdk.samtools.cram.ref.ReferenceContext;
+import htsjdk.samtools.cram.structure.CRAMEncodingStrategy;
+import htsjdk.samtools.cram.structure.CRAMStructureTestHelper;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public class SliceFactoryTest  extends HtsjdkTest {
+
+    @DataProvider(name="emitSliceCoordinateSortedPositive")
+    private Object[][] getEmitSliceCoordinateSortedPositive() {
+        final int MAPPED_REFERENCE_INDEX = 1;
+
+        // NOTE: These tests use, and assume, the default CRAMEncodingStrategy readsPerSlice and
+        // minimumSingleSliceReferenceSize values
+
+        return new Object[][] {
+                // currentRefContextID, numberOfRecordsSeen, nextRecordRefContextID, updatedRefContextID
+
+                // uninitialized state
+                { ReferenceContext.UNINITIALIZED_REFERENCE_ID, 0, MAPPED_REFERENCE_INDEX,  MAPPED_REFERENCE_INDEX },
```

Comment:
Does this DataProvider encompass all possible state transitions?

Reply:
Yes, for coordinate sorted. Running the tests in this file with code coverage indicates there is one branch that itsn't executed, but adding a query-name sorted test will address with that.